### PR TITLE
RFE#701 - Print View using CSS - Rewrite

### PR DESCRIPTION
**Pull request by devenbansod**
Rewrote the implementation of **Print View** Feature. Now using CSS to design the printing of the page.

Signed-off-by: Deven Bansod devenbansod.bits@gmail.com

### CODE REVIEW

**Comment by devenbansod** [comment]
Most of the related discussion can be found at : #1701

**Comment by ibennetch** [comment]
In the other pull request we were discussing whether it was possible to get the exact view count and how to do so.

I didn't have a good answer to that question and was spending some time thinking about it and looking in to it.

Can we use an approximate count immediately?

**Comment by devenbansod** [comment]
I am slightly confused here.

Because at present, the print view of Views does not mention the row - count at all.
[PDF of Print View on Demo Server](https://devenbansod.files.wordpress.com/2015/05/print-view-phpmyadmin-4-5.pdf)

And if we print the whatever is shown before those clicks, user might fall prey to wrong notions.

**Comment by ibennetch** [comment]
I mean in the normal, screen-based view you can see an approximate row count, then if you click it you see the exact row count. 

Let's move discussion to the -devel mailing list. I'll send a message now.

**Comment by ibennetch** [comment]
On the mailing list Madhura points out something I had overlooked; if the user wishes to get an exact row count they can do it manually before pressing print. So we should leave it how it is and not force the user to get the exact view count; just print whatever is displayed in the field whether it's approximate or exact.

Were there any other concerns about this pull request?

File: libraries/Header.class.php [hunk 1]
```diff
@@ -658,6 +658,9 @@ private function _getLinkTags()
             $retval .= '<link rel="stylesheet" type="text/css" href="'
                 . $basedir . 'phpmyadmin.css.php?'
                 . 'nocache=' . $theme_id . $GLOBALS['text_dir'] . '" />';
+            // load Print view's CSS latest, so that it overrides all other CSSs while 'printing'
```
Comment (ibennetch) [inline libraries/Header.class.php]: s/latest/last/

**Comment by ibennetch** [comment]
I think we should move `printview.css` to the theme directory instead of the root; that way theme developers have the option to change the look of a printout.

**Comment by devenbansod** [comment]
Yes. It makes sense to leave it as it is and let the user click if he/she wants to print the real rows. I will make the required changes.

**Comment by devenbansod** [comment]
About the printview.css, that seems a valid point. I did not think of it before. I will move it to both the themes.

**Comment by devenbansod** [comment]
Also, should I implement the 'Get all Row Counts' link in this feature itself, or should I open a new feature-request ticket and then make a new PR ?

**Comment by devenbansod** [comment]
Now, Scrutinizer errored because it could not find the dependency **guzzle**

**Comment by ibennetch** [comment]
Definitely as a new pull request, but it's beyond the scope of the requirements for 701 so you can work on it if you feel you have time as long as it doesn't interfere with your GSoC schedule :)

**Comment by devenbansod** [comment]
Thanks. 
Opened a PR #1709

**Comment by ibennetch** [comment]
I don't know about this guzzle issue, I'll have to look in to it later today.

**Comment by ibennetch** [comment]
Great work on this. I like that it shows your ability to work through completely different parts of the codebase in different capacities, because this touches several different areas.

I hope you are pleased with your work, it looks good.

**Comment by devenbansod** [comment]
Thanks. :)

Also, had made a small error. Corrected in #1711

**Comment by ibennetch** [comment]
@devenbansod can you take a look at https://sourceforge.net/p/phpmyadmin/bugs/4957/ which may be related to this?

**Pull request merged**